### PR TITLE
Integrate Elastica

### PR DESCRIPTION
**Pull request by RUFFY-369**
Modified Dockerfile, README & docker-compose.yml for ros2-elastica addition

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -35,6 +35,21 @@ RUN . /opt/ros/$ROS_DISTRO/setup.sh && \
...
+#default value of arg ELASTICA_ADD when not provided in the --build-arg
+ARG ELASTICA_ADD="false"
+
+# ros2-elastica dependencies
+RUN if [ "${ELASTICA_ADD}" = "false" ]; then\
+      echo 'Not adding Elastica';\
+    else\
+      echo "Adding Elastica " &&\
+      apt-get update &&\
+      apt-get install --no-install-recommends -y tzdata dirmngr gnupg2 build-essential python3-colcon-common-extensions python3-colcon-mixin python3-rosdep python3-vcstool python3-pip  povray python3-tk &&\
```
Comment (mstoelzle) [inline Dockerfile:47]: Please move the installation of `tzdata  dirmngr gnupg2 dirmngr build-essential python3-colcon-common-extensions python3-colcon-mixin python3-rosdep python3-vcstool python3-pip  python3-tk` to line 8. I think the are quite general and should be installed in any case even without elastica

Comment (mstoelzle) [inline Dockerfile:47]: Have you checked if all of them are necessary or if maybe some of them are already included with the ros:foxy image we are building on?

File: src/overlay.repos [hunk 2]
```diff
@@ -8,4 +8,7 @@ repositories:
         url: https://github.com/tud-cor-sr/ros2-vtem_control
     mocap_optitrack:
         type: git
-        url: https://github.com/tud-cor-sr/ros2-mocap_optitrack.git
\ No newline at end of file
+        url: https://github.com/tud-cor-sr/ros2-mocap_optitrack.git
+    elastica:
```
Comment (mstoelzle) [inline src/overlay.repos:12]: I think elastica should only be cloned if the argument is set to true. So please move the elastica repo to its own `elastica.repos` file, any basically copy lines 27-36 in the Dockerfile for the `elastica.repos` and only run it when the argument is set to true

**Review by mstoelzle — CHANGES_REQUESTED** [review]
Hi Prakarsh, it looks very good already! I only have two (minor) points of improvements, then we can merge. Please confirm again that it builds without any errors before we merge

**Comment by RUFFY-369** [comment]
> Hi Prakarsh, it looks very good already! I only have two (minor) points of improvements, then we can merge. Please confirm again that it builds without any errors before we merge

Yeah it builds without any errors. I tested the scripts as well. Let me make the requested  changes, test again & commit.

File: Dockerfile (hunk 1 above)
Comment (RUFFY-369) [reply Dockerfile:47]: Yeah those were in my Dockerfile. Let me shift these general dependencies with others.

Comment (RUFFY-369) [reply Dockerfile:47]: Some of them I'm sure of are necessary like povray, python3-tk for visualisation. Let me check the others if they are included with the base image of ros:foxy.

Comment (mstoelzle) [reply Dockerfile:47]: Did your Dockerfile build on `ros:foxy` as well?

File: src/overlay.repos (hunk 2 above)
Comment (RUFFY-369) [reply src/overlay.repos:12]: Yeah, thanks for pointing out. I forgot to set cloning elastica depending on the arg value. I'll fix that in the next commit.

File: docker-compose.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@ services:
         context: .
         dockerfile: Dockerfile
       environment:
```
Comment (mstoelzle) [inline docker-compose.yml:8]: @RUFFY-369 Something I forgot earlier: Please also add the argument to activate / deactivate elastica here so its easily discoverable for people working with docker-compose

File: Dockerfile (hunk 1 above)
Comment (RUFFY-369) [reply Dockerfile:47]: Yes

File: docker-compose.yml (hunk 3 above)
Comment (RUFFY-369) [reply docker-compose.yml:8]: Oh! Yeah that I'll add in the file

File: Dockerfile (hunk 1 above)
Comment (RUFFY-369) [reply Dockerfile]: I have removed all of them except the last three (as they were there in the base image  of ros:foxy)

Comment (RUFFY-369) [reply Dockerfile]: I removed all of them except the last three as they were in the base image of `ros:foxy` & removing them didn't affect the original functioning

File: src/overlay.repos (hunk 2 above)
Comment (RUFFY-369) [reply src/overlay.repos]: I did that in the latest commit

File: docker-compose.yml (hunk 3 above)
Comment (RUFFY-369) [reply docker-compose.yml:11]: Done in the latest commit

**Comment by RUFFY-369** [comment]
Hi @mstoelzle , I have pushed the requested changes. Please review them

**Review by mstoelzle — APPROVED** [review]

**Pull request merged**